### PR TITLE
Add fuel service information

### LINKED ISSUE #1827: New quest: Ask if a fuel station offers self service
### General
Affected tag(s) to be modified/added: [self_service](https://wiki.openstreetmap.org/wiki/Key:self_service)
Question asked: **Does this fuel station offer self service?** (yes / no / only)

### Checklist
Checklist for quest suggestions (see guidelines):
- [x] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [x] 🐿️ Easily answerable by everyone from the outside but a survey is necessary 
- [x] 💤 Not an overwhelming percentage of elements have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of elements (Worth the effort)

### Ideas for implementation

Like most of the quests there's a question and three option: yes, no or only when the fuel station only offers self service.

**Element selection:**
 ```
/*
This has been generated by the overpass-turbo wizard.
The original search was:
“amenity=fuel and self_service!=*”
*/
[out:json][timeout:25];
// gather results
(
  // query part for: “amenity=fuel and self_service!=*”
  node["amenity"="fuel"]["self_service"!~".*"]({{bbox}});
  way["amenity"="fuel"]["self_service"!~".*"]({{bbox}});
  relation["amenity"="fuel"]["self_service"!~".*"]({{bbox}});
);
// print results
out body;
>;
out skel qt;
```

**Metadata needed:** It would be useful in Italy since not every fuel station has self service and a lor of them aren't tagged as self service anyway.

**Proposed GUI:**   (may make an image later)

|     | **Does this fuel station offer self service?** |                          |
|-----|------------------------------------------------|--------------------------|
| yes | no                                             | only offers self service |
|     |                                                |                          |
|     |                                                |                          |

### PR DESCRIPTION
**Pull request by naposm**
This is the pull request for #1827. I'll just add the updated screenshots here.

<details>
<summary>SCREESHOTS</summary>
<b>Self service quest</b>: <br>
 

<b>Full service quest</b><br>

</details>

closes #1827

### CODE REVIEW

**Comment by andrewharvey** [comment]
I think disable for `AU` too. I've never seen a petrol station where a staff will pump fuel for you. They are all self service only, so not something we need a quest for here.

@dbdean any opinion?

**Comment by matkoniecz** [comment]
Note that `full_service` tag is barely used, just 200 uses worldwide so it may be a good idea to check on tagging mailing list is it a good idea to start adding it on all fuel stations (is it good enough for SC quest?).

Also, is it meaning even really clear?

**Comment by naposm** [comment]
> Note that `full_service` tag is barely used, just 200 uses worldwide so it may be a good idea to check on tagging mailing list is it a good idea to start adding it on all fuel stations (is it good enough for SC quest?).
Yes, I'm taking this in consideration, that's why it's a draft.

> Also, is it meaning even really clear?
The term is on the [Cambridge dictionary](https://dictionary.cambridge.org/dictionary/english/full-service), so I think it is. Maybe a better term could be used for the tag, but for the quest I think that's what it should ask.

**Comment by dbdean** [comment]
> I think disable for `AU` too. I've never seen a petrol station where a staff will pump fuel for you. They are all self service only, so not something we need a quest for here.
> 
> @dbdean any opinion?

Well, we do have one in Armidale that pump fuel for you. My wife loves it, but it's to weird for me so I avoid using them :) I don't recall seeing one before in any other place I've lived though.

**Comment by westnordost** [comment]
Just a short comment on the icon:
The upmost one looks oddly out of focus (left car, right person). The one with the blue man a little bit small - maybe the fuel pump or the man could be made larger, so that they are not next to each other but a little bit in front of each other

**Comment by trigpoint** [comment]
Whilst I am aware that one or two US states make self serve illegal, this should be disabled for the United Kingdom and probably most of mainland Europe where unstaffed pay at pump is very much the norm.

**Comment by matkoniecz** [comment]
What was outcome of full_service discussion? From cursory look it appears that meaning of that term is quite unclear and/or duplicate of `self_service`.

https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dfuel#Service mentions it right now but without any hint about meaning.

**Comment by letypequividelespoubelles** [comment]
> this should be disabled for the United Kingdom and probably most of mainland Europe where unstaffed pay at pump is very much the norm.

+1 It should be disabled too for non-continental French territories (such as Réunion, Guyane, Guadeloupe, Martinique, Mayotte) where, contrary to mainland France, all pumps are staffed.

**Comment by naposm** [comment]
> What was outcome of full_service discussion? From cursory look it appears that meaning of that term is quite unclear and/or duplicate of `self_service`.
> 
> https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dfuel#Service mentions it right now but without any hint about meaning.

There were a lot of great ideas and opinions, but there was not a precise outcome, the full service tag is both praised and hated as I can understand, because there are only few countries where this could be useful. 
So I was thinking the opposite, since I don't want to make another Italy-centric quest, I was thinking about some options:
1. Make a survey and understand which countries could actually benefit from this quest (so the `full_service` tag is considered ok) 
2. Only ask for `self_service` in selected countries (same as above but `full_service` is assumed, with `self_service=only` being a option.
3. Change the way this is asked, ask which kind of services the fuel station offers (cards with images and full, self service, both) 
4. Just remove the quest for every country (close the PR)

**Comment by westnordost** [comment]
Sorry for not getting back at you earlier.

I'd say, keep it simple first. Option 2 sounds good:
- only ask in countries where some fuel stations are self service and others are not-self-service (Italy?)
- ask only for `self_service` . The wording "does X offer self service?" sounds a a bit odd. They offer nothing if there is self service. Didn't think about htis a lot yet if there is a better wording that makes it possible to answer with yes/no. If not, it may be better to either have a picture select or a radio button select (like in `AddInternetAccessForm`)

**Comment by naposm** [comment]
No problem, don't worry, I understand that everyone here has their own life too and StreetComplete is a very big project to deal with, so don't have to apologize 🙂.

To understand which countries could benefit from the quest we could ask on the forum if you agree, I think that you being German can confirm it's not useful for you, on the other hand I can say that for me here in Italy would be useful (I've seen some gas stations which were closed at night, since they are full service only, also in some major cities, usually very small ones). 
As for the wording, not being a native English speaker is a barrier for me, but I am open to suggestions, if I can find a better wording I'll post it here. The image suggestion is a very good idea, since we already have the full service icon which would be lost, we could integrate it and make images to ask something like:

"Can you refuel using self service at *GAS STATION*?" (please help me with the wording)

*image with the stickman pumping their own gas*
"It is self service only" (tag: self_service=only)

*image with the stickman pumping their own gas and a worker*
"Self service is possible" (tag: self_service=yes)

*image with a big red circle (the one used for no pedestrians allowed) on the stickman pumping their own gas*
"No, you can't use self service here"

By the way, congratulations for your State of the Map talk, you did a very good job! 🥳

**Comment by timothywashere** [comment]
> 

Many rural petrol stations in the UK are Full Service only. You're not allowed to "pump your own gas".
I think the question has value in the UK even if all Urban stations are Self-service.

It's also important that the wording does not confuse users with the difference between Self-service and Pay@Pump, which is becoming very popular in the UK.

**Comment by naposm** [comment]
Thanks for the information @timothywashere, sorry if I noticed your comment just now! 

By the way I was thinking about removing the full service question altogether and just leaving the self service one, that I think has more relevant use cases, as usually it's obvious that where there's no self service, there must be full service (your statement is an example).

I would really like to show it everywhere, as what you said is basically my use case for Italy. I'd like to find a way to show the quest only to relevant gas stations all over the world except the few cases where it's restricted by law (example in USA). I don't know if there's a way to find elements near the gas station or if that would be useful, but that would be great. 

I proposed this quest when there was the lockdown and maps like Ca reste ouvert only showed gas station with the self service tag as open. I still think that having this information could be useful for some purposes in the future, but implementing it doesn't seem as simple as I initially thought it to be...

**Comment by westnordost** [comment]
Just self service yes / no should be an easy yes/no question and would be very easy to implement. The "complicated" part would be to research in which countries it should be enabled. I.e. where do both exist. It is okay to only start with 1-few countries, it is not necessary to research for the whole world.

**Comment by mnalis** [comment]
> I proposed this quest when there was the lockdown and maps like Ca reste ouvert only showed gas station with the self service tag as open. I still think that having this information could be useful for some purposes in the future, but implementing the quest doesn't seem as simple as I initially thought it to be...

Oh, wait, why would covid-19 lockdown be indicative of that? It seems like they/you consider `self_service=yes` to mean "without [cashiers](https://en.wikipedia.org/wiki/Cashier)"?

My reading of [self_service=yes](https://wiki.openstreetmap.org/wiki/Key:self_service) is that it is about whether the **service itself is provided by other humans**, or do you do it yourself. I.e. in case of fuel pump, do you put the nozzle into your car reservoir yourself, or if employee does it for you. But **not** if the human employee is involved with the payment for the goods/services.

So I interpret it the same as it is with other tags listed in `self_service=*` (like `canteen`, `restaurant`, `fast_food`... if `self_service=yes` you will grab food yourself, but there is still **usually** human cashier at the end of the line to whom you pay for the food)

So I think `self_service=yes` is a bad proxy for indicating there is no human interaction needed at the location complete the transaction. For example, in Croatia, almost all fuel stations are `self_service=yes`, but most of them are also `automated=no` -- ie. you'll **fill the tank yourself**, but then go inside to **pay to the human cashier** (which won't help you much in reducing COVID-19 transmissions, as there would still be lots of human interactions)


I think [automated=yes](https://wiki.openstreetmap.org/wiki/Key:automated) is the closest tag to indicate whether you'll have to interact with other humans to finish the job or not. If we want to indicate fully automated fuel station without any humans onboard, it would probably need to be marked `self_service=only` + `automated=yes`.

(note: In this review I do not even analyze `full_service` as you're not sure if you'll implement it)

---

TL;DR: please make clear a distinction whether the quest is about if the users needs to perform service themselves (ie. `self_service=yes`) or whether it is about whether there are any cashiers you need to interact with or not (ie. `automated=no/yes`). Or does the quest intent to cover both of those meanings?

**Comment by naposm** [comment]
> The "complicated" part would be to research in which countries it should be enabled. I.e. where do both exist. 

Yes, that was exactly what I meant with difficult, not the code per se, that fortunately, I must give credit for you and all contributors who worked on it, is very simple to write for basic quests even for a beginner like me 😁


> Oh, wait, why would covid-19 lockdown be indicative of that? 

I don't know why but it was like that, if I recall correctly there was a law here in Italy which demanded self service gas station (I'm not sure, might be wrong, more than a year passed), so maybe there was a similar one in France.

> It seems like they/you consider `self_service=yes` to mean "without [cashiers](https://en.wikipedia.org/wiki/Cashier)"? 
   My reading of [self_service=yes](https://wiki.openstreetmap.org/wiki/Key:self_service) is that it is about whether the **service itself is provided by other humans**, or do you do it yourself. I.e. in case of fuel pump, do you put the nozzle into your car reservoir yourself, or if employee does it for you. But **not** if the human employee is involved with the payment for the goods/services.
   So I interpret it the same as it is with other tags listed in self_service=* (like canteen, restaurant, fast_food... if self_service=yes you will grab food yourself, but there is still usually human cashier at the end of the line to whom you pay for the food)

Yes, usually I meant self service without cashier, but you're right I have sometimes found self service where you paid to a cashier but pumped your own gas... You just made me think about it, I had never considered this 🤦‍‍♂️

> I think [automated=yes](https://wiki.openstreetmap.org/wiki/Key:automated) is the closest tag to indicate whether you'll have to interact with other humans to finish the job or not. If we want to indicate fully automated fuel station without any humans onboard, it would probably need to be marked `self_service=only` + `automated=yes`.

Mh... Maybe we need more than a quest. I just assumed `automated=yes` because I did not think about having self service with payment to a cashier, I just forgot that option, my bad. This quest would only ask for self service, but the automated quest could be added I think, maybe with another quest? If you have any ideas let me know 🤔
 
> (note: In this review I do not even analyze `full_service` as you're not sure if you'll implement it)

You did well, I think that dealing with non standard tags might be dangerous at the moment... It's not very clear its utility and meaning so I think it's better to avoid it for now (to be honest I'm starting to doubt about its utility myself).

Thanks by the way for bringing to light the matter!

**Comment by smichel17** [comment]
I think if a station has self service, and accepts credit cards, it can be *mostly* assumed that you do not have to interact with the cashier. At least in the US.

On January 9, 2022 11:44:11 PM UTC, naposm ***@***.***> wrote:
>> The "complicated" part would be to research in which countries it should be enabled. I.e. where do both exist. >
>
>
>Yes, that was exactly what I meant with difficult, not the code per se, that fortunately, I must give credit for you and all contributors who worked on it, is very simple to write for basic quests even for a beginner like me 😁
>
>
>
>
>
>> Oh, wait, why would covid-19 lockdown be indicative of that? >
>
>
>I don't know why but it was like that, if I recall correctly there was a law here in Italy which demanded self service gas station (I'm not sure, might be wrong, more than a year passed), so maybe there was a similar one in France.
>
>
>
>> It seems like they/you consider `self_service=yes` to mean "without [cashiers](https://en.wikipedia.org/wiki/Cashier)"? >
>   My reading of [self_service=yes](https://wiki.openstreetmap.org/wiki/Key:self_service) is that it is about whether the **service itself is provided by other humans**, or do you do it yourself. I.e. in case of fuel pump, do you put the nozzle into your car reservoir yourself, or if employee does it for you. But **not** if the human employee is involved with the payment for the goods/services.
>
>   So I interpret it the same as it is with other tags listed in self_service=* (like canteen, restaurant, fast_food... if self_service=yes you will grab food yourself, but there is still usually human cashier at the end of the line to whom you pay for the food)
>
>
>
>Yes, usually I meant self service without cashier, but you're right I have sometimes found self service where you paid to a cashier but pumped your own gas... You just made me think about it, I had never considered this 🤦‍‍♂️
>
>
>
>> I think [automated=yes](https://wiki.openstreetmap.org/wiki/Key:automated) is the closest tag to indicate whether you'll have to interact with other humans to finish the job or not. If we want to indicate fully automated fuel station without any humans onboard, it would probably need to be marked `self_service=only` + `automated=yes`.
>
>
>
>Mh... Maybe we need more than a quest. I just assumed `automated=yes` because I did not think about having self service with payment to a cashier, I just forgot that option, my bad. This quest would only ask for self service, but the automated quest could be added I think, maybe with another quest? If you have any ideas let me know 🤔
>
> >
>> (note: In this review I do not even analyze `full_service` as you're not sure if you'll implement it)
>
>
>
>You did well, I think that dealing with non standard tags might be dangerous at the moment... It's not very clear its utility and meaning so I think it's better to avoid it for now (to be honest I'm starting to doubt about its utility myself).
>
>
>
>Thanks by the way for bringing to light the matter!
>
>
>
>
>
>-- >
>Reply to this email directly or view it on GitHub:
>
>https://github.com/streetcomplete/StreetComplete/pull/2822#issuecomment-1008448261
>
>You are receiving this because you are subscribed to this thread.
>
>
>
>Message ID: ***@***.***>
-- 
To respect your time, I try to write short, functional emails.

**Comment by westnordost** [comment]
In Germany, all stations have self service. You still need to go into the shop and pay for the gasoline.

**Comment by naposm** [comment]
<div style="float:left">

</div>

Hi everyone! I removed the full service quest and made some changes to the full service one.

File: app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    override fun getTitle(tags: Map<String, String>) : Int {
+        val hasName = tags.containsKey("name")
+        val hasBrand = tags.containsKey("brand")
+        return when {
+            hasName || hasBrand ->  R.string.quest_fuelSelfService_name_title
+            else ->                 R.string.quest_fuelSelfService_title
+        }
+    }
+
+    override val enabledInCountries = NoCountriesExcept("IT")
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt:34]: Because of https://github.com/streetcomplete/StreetComplete/pull/2822#issuecomment-977850275, maybe it should be enabled in the UK?

```suggestion
    override val enabledInCountries = NoCountriesExcept("IT", "UK")
```

File: res/graphics/quest icons/fuel_self_service.svg [hunk 2]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (FloEdelmann) [inline res/graphics/quest icons/fuel_self_service.svg:1]: I think the icon is a bit too busy, but I'm fine with it.

However, could you run it through `svgo` or save it as "Optimized SVG" in Inkscape to clean up unneeded stuff, please?

File: app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt (hunk 1 above)
Comment (naposm) [reply app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt:34]: Mh, I did not understand if it's really useful or not as it is now, @timothywashere what do you think?

File: res/graphics/quest icons/fuel_self_service.svg (hunk 2 above)
Comment (naposm) [reply res/graphics/quest icons/fuel_self_service.svg:1]: >  I think the icon is a bit too busy, but I'm fine with it.

I'm not a big fan neither, I'm not good at designing things 🙄 By the way I can't find on the web how to mask the icon properly (I noticed that my image has part of the icon outside the circle when viewed in Firefox, but it's ok in the app.

> However, could you run it through svgo or save it as "Optimized SVG" in Inkscape to clean up unneeded stuff, please?

Sure!

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -604,6 +604,8 @@ Otherwise, you can download another keyboard in the app store. Popular keyboards
     <string name="quest_construction_steps_title">Are these steps completed?</string>
     <string name="quest_construction_generic_title">Is this way completed?</string>
     <string name="quest_construction_building_title">Is this building completed?</string>
+    <string name="quest_fuelSelfService_title">Can you pump your own gas at this fuel station?</string>
```
Comment (westnordost) [inline app/src/main/res/values/strings.xml:607]: Sounds like you can bring your own gas to pump here.

**Review by westnordost — CHANGES_REQUESTED** [review]
Thank you for your persistence! The wording sounds a bit off, otherwise it looks good.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (riQQ) [reply app/src/main/res/values/strings.xml:607]:
```suggestion
    <string name="quest_fuelSelfService_title">Can you pump gas on your own at this fuel station?</string>
```

**Comment by naposm** [comment]
> Thank you for your persistence! 

Hahah I hope this is a good thing 😂
I just love the app so I'm happy to give a contribute somehow 😁

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (naposm) [reply app/src/main/res/values/strings.xml:607]: Mh I don't know how to write it because English isn't my mother tongue 😕 I trust @riQQ 😅

This is a preview (I also changed the icon):

**Comment by westnordost** [comment]
The icon is better now!

**Comment by westnordost** [comment]
That's ok,  I'll take care of it after merge

On 17 January 2022 10:24:49 CET, naposm ***@***.***> wrote:
>Thanks! The problem with the mask is still there unfortunately. I can't
>find any tutorial on how to make a mask like yours where the icon
>doesn't show the masked elements in Firefox 😔>
>>
>-- >
>Reply to this email directly or view it on GitHub:>
>https://github.com/streetcomplete/StreetComplete/pull/2822#issuecomment-1014307460>
>You are receiving this because you were mentioned.>
>>
>Message ID:
***@***.***>

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

File: app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt (hunk 1 above)
Comment (timothywashere) [reply app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt:34]: > Mh, I did not understand if it's really useful or not as it is now, @timothywashere what do you think?

I think it's useful and would like to see it.

Despite the fact that I used the phrase earlier, "Pump your own gas" isn't really appropriate in British English, as an Americanism it would offend a lot of people. I'm not sure what is a succinct correct phrase, "Dispense your own fuel?" I suppose.

If anyone objects to this quest in the UK an option would be to only shown the quest for small or unknown brands. You wouldn't get self-dispense at BP/Shell/Esso/Tesco/Morrisons/Asda/Sainsburys. But I am happy to answer this quest for all petrol stations so don't add that functionality unless someone else complains.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (westnordost) [reply app/src/main/res/values/strings.xml]: I propose "Can you pump gas by yourself at this fuel station?".

File: app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    override val elementFilter = """
+        nodes, ways with
+          amenity = fuel
+          and !self_service
+          and !automated
+    """
+    override val commitMessage = "Add self service information to fuel station"
+    override val wikiLink = "Tag:amenity=fuel"
+    override val icon = R.drawable.ic_fuel_self_service
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt:21]: The icon should be called `ic_quest_fuel_self_service` to be consistent with other quests.

**Review by FloEdelmann — DISMISSED** [review]

File: app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt (hunk 1 above)
Comment (timothywashere) [reply app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt:34]: Also one pedantic query. Should it be "Can you.." or "Must you.." . Can implies a choice that isn't there (In the UK anyway).
For self service stations it's usually required that you operate the pump yourself, unless you are disabled and may be able to ask for help.
If it's not self service, then you are forbidden from operating the pump.

Comment (naposm) [reply app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt:34]: @timothywashere Thank you for the answer, not being English I'm unaware of what's correct and what isn't and can't appreciate the nuances between two wordings, if you can suggest a question for the tag I'd be very glad.

I added UK by the way, thank you!

Comment (naposm) [reply app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt:34]: > If anyone objects to this quest in the UK an option would be to only shown the quest for small or unknown brands. You wouldn't get self-dispense at BP/Shell/Esso/Tesco/Morrisons/Asda/Sainsburys.

I don't know how this could be handled, maybe just remove all fuels with brand tags altogether?

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (naposm) [reply app/src/main/res/values/strings.xml]: @timothywashere suggested something along the line of "Dispense your own fuel?"

File: app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+class AddFuelSelfService : OsmFilterQuestType<Boolean>() {
+
+    override val elementFilter = """
+        nodes, ways with
+          amenity = fuel
+          and !self_service
+          and !automated
+    """
+    override val commitMessage = "Add self service information to fuel station"
+    override val wikiLink = "Tag:amenity=fuel"
```
Comment (naposm) [inline app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt:20]: Is `Tag:amenity=fuel` correct or should it be [`Key:self_service`](https://wiki.openstreetmap.org/wiki/Key:self_service)? 🤔

Comment (westnordost) [reply app/src/main/java/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt:20]: better  Key:self_service

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (timothywashere) [reply app/src/main/res/values/strings.xml]: "Do you dispense your own fuel at .... ?"

"Do you operate the pump yourself at ...?"

In terms of absolute clarity I think maybe:
"Are the pumps at ... operated by the customer?"

Comment (westnordost) [reply app/src/main/res/values/strings.xml]: I also asked in OSM slack US channel #language: 

image

Comment (timothywashere) [reply app/src/main/res/values/strings.xml]: Yes, undoubtably "Pump your own gas" is correct in American English.

**Comment by westnordost** [comment]
I think everything was OK with the mask, just that the viewer you used doesn't properly handle masks.

For this reason and because masks are only supported at a certain API level in Android (and otherwise they automatically get rasterized by the build process), I ususally "bake" the masks in. I did this now too with your icon. The way I do it in Inkscape: select both the circular background and the thing that should be masked by that circular background, select Path->Intersection (Ctrl+*). Copy the result and undo the last operation. Then, paste: Replace the original with the intersection result. Not sure if there is any mor estraightforward way to do it.

**Comment by naposm** [comment]
That was a surprise, thank you very much Tobias for merging! And thanks to FloEdelmann, timothywashere, mnalis, riQQ and anyone who helped out here and in the linked issues (too many people to thank)!

(small note @westnordost : I did not update `wikiLink` to `Key:self_service` because I did not check here until now, would you like me to submit a PR for that?)

P.S. Thank you for letting me know the Inkscape trick, I think it will be very useful to me, I looked it everywhere, I hate having that mask problem, so thank you very much!

**Comment by westnordost** [comment]
I already did it myself. FYI this will go into v40. Tomorrow or so I'll release v39.0

**Comment by naposm** [comment]
That's nice, thanks!

**Comment by matkoniecz** [comment]
> Many rural petrol stations in the UK are Full Service only. You're not allowed to "pump your own gas".

you mean United Kingdom here, right?

`app/src/androidMain/kotlin/de/westnordost/streetcomplete/quests/fuel_service/AddFuelSelfService.kt` likely should be changed from `NoCountriesExcept("IT", "UK")` to `NoCountriesExcept("IT", "GB")`
